### ISSUE TITLE
Logowanie danych nowych użytkowników

### ISSUE THREAD

**Issue by MacDada** [dudekm/blog#21]
Przy tworzeniu nowego użytkownika zapisuj jego dane do pehapowego error loga.

Najprostsza opcja to po prostu zrobić `error_log('Nowy użytkownik: '.$user->getUsername());` i na początek na tej zasadzie właśnie zrób.

- [ ] logowanie username użytkownika do error loga (w repozytorium)

Załóżmy, że w przyszłości chcemy, żeby logowanie było bardziej skomplikowane, wydzielamy więc dedykowany obiekt:

- [ ] utwórz klasę `Logger`, która będzie miała metodę `log($message)`
- [ ] przekazuj obiekt `Logger`a do repozytorium, żeby repo z niego skorzystało

Ponieważ nie chcemy wynajdować koła na nowo, okazuje się, że ustalono standard, który mówi jak konstruować klasy logujące.

- [ ] zaciągnij composerem https://github.com/php-fig/log
- [ ] w repozytorium wymagaj przekazania `Psr\Log\LoggerInterface`
- [ ] `Logger` niech spełnia `LoggerInterface`

**Comment by dudekm** [comment]
@MacDada do czego potrzebna jest ta klasa? Do zapisywania informacji czy dodano użytkownika czy też nie?

**Comment by MacDada** [comment]
M.in. do debugowania. Później podepniesz error_handler i będziesz miał wszystkie błędy w pliku z logiem.

1. Ktoś wchodzi na stronę, występuję jakiś błąd.
2. Ten błąd powinien trafić do logów.
3. Ty potem możesz przeczytać plik loga, żeby dowiedzieć się, gdzie są błędy i je naprawić.

**Comment by dudekm** [comment]
@MacDada a wyjątki? :) mówiłeś, że tu przechwytuje się błędy więc przypuszczam, że również można je w tym miejscu zapisać. Czy wtedy tu się używa też tego error_loga ?

**Comment by MacDada** [comment]
tak, można automatycznie przechwytywać zarówno errory jak i wyjątki: https://secure.php.net/manual/en/function.set-error-handler.php

**Comment by dudekm** [comment]
@MacDada klasa Logger będzie modelem? Metoda log($message) ma wywoływać error_loga i zwracać coś?

**Comment by MacDada** [comment]
@dudekm Na początek utwórz sobie nowy katalog, np `Logging` i w niej klasę `ErrorLogLogger`.
Metoda `log()` nie musi nic zwracać, po prostu ma wrzucić wiadomość do ErrorLoga.
